methods.py

```diff
@@ -1048,6 +1048,40 @@ def progress_finish(target, source, env):
     AlwaysBuild(progress_finish_command)
 
 
+def validate_target_filename(suffix):  # Check if target file can be written to.
+    if os.access("bin/godot" + suffix, os.F_OK) and not os.access("bin/godot" + suffix, os.W_OK):
+        suffixes = suffix.split(".")  # Split suffix string so we can add suffix number later
+        extension = suffixes.pop()  # Store last suffix, should be the file extension.
```

Comment:
```suggestion
        name, _, extension = suffix.rpartition(".")  # Split suffix string so we can add suffix number later
```

example of `rpartition()` below. The `name` will be returned as string, so no need to do `'.'.join(name)` later again
```
'godot'.rpartition('.')
>>> ('', '', 'godot')

'foo.bar.godot'.rpartition('.')
>>> ('foo.bar', '.', 'godot')
```

Reply:
Awesome, thanks for the input. I've implemented your changes.